Sources/Testing/Issues/KnownIssue.swift

```diff
@@ -8,25 +8,53 @@
 // See https://swift.org/CONTRIBUTORS.txt for Swift project authors
 //
 
-/// Combine an instance of ``KnownIssueMatcher`` with any previously-set one.
-///
-/// - Parameters:
-///   - issueMatcher: A function to invoke when an issue occurs that is used to
-///     determine if the issue is known to occur.
-///   - matchCounter: The counter responsible for tracking the number of matches
-///     found with `issueMatcher`.
-///
-/// - Returns: A new instance of ``Configuration`` or `nil` if there was no
-///   current configuration set.
-private func _combineIssueMatcher(_ issueMatcher: @escaping KnownIssueMatcher, matchesCountedBy matchCounter: Locked<Int>) -> KnownIssueMatcher {
-  let oldIssueMatcher = Issue.currentKnownIssueMatcher
-  return { issue in
-    if issueMatcher(issue) || true == oldIssueMatcher?(issue) {
-      matchCounter.increment()
-      return true
+/// A type that represents an active `withKnownIssue()` call and any parent calls.
+///
+/// A stack of these is stored in `KnownIssueContext.current`. The stack is
+/// mutated by calls to `withKnownIssue()`.
+struct KnownIssueScope: Sendable {
+  /// Determine if an issue is known to this scope or any of its ancestor
+  /// scopes.
+  ///
+  /// Returns `nil` if the issue is not known.
+  var match: @Sendable (Issue) -> Issue.KnownIssueContext?
```

Comment:
Maybe `matcher` (-er) instead?

Reply:
I was thinking this name reads better at the call site:

```swift
if let context = scope.match(issue) {
```

Reply:
I'd find it nicer if we used a `typealias` for the function type, because then we could document the semantics of the matcher function itself and more clearly indicate the `- Return:` and other fields. E.g.

```swift
/// A function which ...
///
/// - Parameters:
///   - issue: ...
///
/// - Returns: ...
typealias Matcher = @Sendable (_ issue: Issue) -> Issue.KnownIssueContext?
```

Reply:
Renamed to `matcher` and added the typealias.